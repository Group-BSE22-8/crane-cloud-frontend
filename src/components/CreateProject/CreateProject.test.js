/* eslint-disable no-undef */
import React from "react";
import { shallow } from "enzyme";
import CreateProject, { mapStateToProps } from "./";

const CreateProjectProps = {
  clusters: { log1: 1, log2: 2 },
  match: { params: { projectId: "1" } },
  clearUpdateClusterState: jest.fn(),
  updateCluster: jest.fn(),
  data: { beta: true },
  clearState: jest.fn(),
  getClustersList: jest.fn(),
  clearAddProjectState: jest.fn(),
};
// {} ={}
describe("Testing the App Metrics Page component", () => {
  const WrapperCreateProject = CreateProject.WrappedComponent;
  const CreateProjectComponent = shallow(<WrapperCreateProject {...CreateProjectProps} />);
  it("should match the snapshot for CreateProject after adding props", () => {
    CreateProjectComponent.setProps(CreateProjectProps);
    expect(CreateProjectComponent).toBeDefined();
  });
  it("matchs the CreateProject component snapshot", () => {
    expect(CreateProjectComponent).toMatchSnapshot();
  });
});

describe("Testing the exported mapstate to props and dispatch", () => {
  it("matches the Appmetricspage mapstostate", () => {
    expect(
      mapStateToProps({
        clustersReducer: {
          clusters: [],
        },
        user: { data: [] },
        addProjectReducer: {
          isAdding: false,
          isAdded: false,
          message: "",
          errorCode: "",
        },
      })
    ).toEqual({
      clusters: [],
      data: [],
      isAdding: false,
      isAdded: false,
      message: "",
      errorCode: "",
    });
  });
});
