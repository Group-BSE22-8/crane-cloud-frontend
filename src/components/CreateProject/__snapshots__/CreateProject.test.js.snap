// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the App Metrics Page component matchs the CreateProject component snapshot 1`] = `ShallowWrapper {}`;
